### PR TITLE
merge v0.23.29 -> main

### PR DESCRIPTION
**Pull request by samspills**
the only conflicts I had to resolve were 
- the changelog
- the scala versions in `project/HttpPlugin.scala`

but then I had to push a fix to remove an unused import from a test suite that hasn't been touched in years. I'm not sure what has changed there?

### CODE REVIEW

File: server/shared/src/test/scala/org/http4s/server/middleware/HttpsRedirectSuite.scala [hunk 1]
```diff
@@ -23,7 +23,6 @@ import cats.syntax.all._
 import org.http4s.Uri.Authority
 import org.http4s.Uri.RegName
 import org.http4s.Uri.Scheme
-import org.http4s._
```
Comment (samspills) [inline server/shared/src/test/scala/org/http4s/server/middleware/HttpsRedirectSuite.scala:26]: This is the import I had to remove to appease tests, and I'm not sure what change has caused it to fail here (but be okay in series/0.23)

**Review by danicheg — APPROVED** [review]

**Comment by danicheg** [comment]
@samspills Wouldn't you mind driving the publishing of a subsequent release milestone?

**Comment by samspills** [comment]
@danicheg I can do that! 
I haven't done a milestone release, now that this has merged (thanks Ross!) is the process the same as [described for 0.23.x here](https://github.com/http4s/http4s/blob/series/0.23/RELEASING.md#v023xx-releases)??

**Comment by danicheg** [comment]
@samspills Yeah, it's pretty much the same with the key difference — we should use the `main` branch instead of `series/0.23` and create a 'discussion' in Github when creating a release in Github (it's a checkbox).

**Comment by samspills** [comment]
Okay, I will get to that some time today :D

**Pull request merged**